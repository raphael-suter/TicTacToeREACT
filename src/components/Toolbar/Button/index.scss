@import "../../../styles/colors.scss";

.toolbar-button {
  margin-left: 0.8rem;
  display: flex;
  align-items: center;
  background: $mainColor;
  border: none;
  cursor: pointer;

  span {
    font-size: 24px;
    color: $lightColor;
  }
}
