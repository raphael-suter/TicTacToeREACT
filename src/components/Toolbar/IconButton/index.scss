@import "../../../styles/colors.scss";

.iconbutton {
  margin-left: 0.8rem;
  display: flex;
  align-items: center;
  background: $mainColor;
  border: none;
  cursor: pointer;

  &__icon {
    font-size: 24px;
    color: $lightColor;
  }
}
