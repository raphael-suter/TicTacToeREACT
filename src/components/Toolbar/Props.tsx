export default interface Props {
  title: string;
  score: string;
  buttonIcon: string;
  buttonOnClick: () => void;
}
