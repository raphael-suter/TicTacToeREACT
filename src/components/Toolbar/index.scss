@import "../../styles/colors.scss";

.toolbar {
  display: flex;
  align-items: center;
  background-color: $mainColor;
  box-shadow: 0 0 2px 0 $shadow;

  &__title {
    flex: 1;
    padding: 0.8rem 1.2rem;
    font-size: 24px;
    font-weight: 100;
    color: $lightColor;
  }

  &__score {
    font-size: 20px;
    font-weight: 700;
    color: $lightColor;
  }

  &__buttongroup {
    flex: 1;
    display: flex;
    justify-content: flex-end;
    padding: 0.8rem 1.2rem;
  }
}

@media (max-width: 800px) {
  .toolbar {
    flex-wrap: wrap;

    &__score {
      order: 1;
      flex-basis: 100%;
      padding: 0.6rem 1.5rem;
      text-align: center;
      background: $mediumColor;
      color: $darkColor;
    }
  }
}
