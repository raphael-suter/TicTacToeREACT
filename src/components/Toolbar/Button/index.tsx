import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ icon, onClick }: Props) => (
  <button className='toolbar-button' onClick={onClick}>
    <span className='material-icons'>{icon}</span>
  </button>
);
