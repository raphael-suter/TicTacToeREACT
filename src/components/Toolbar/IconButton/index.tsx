import React from 'react';
import styled from 'styled-components';
import Props from './Props';

const Button = styled.button`
  margin-left: 1rem;
  display: flex;
  align-items: center;
  background: ${({ theme }) => theme.mainColor};
  border: none;
  cursor: pointer;
`;

const Icon = styled.span`
  font-size: 24px;
  color: ${({ theme }) => theme.lightColor};
`;

export default ({ icon, onClick }: Props) =>
  <Button onClick={onClick}>
    <Icon className='material-icons'>{icon}</Icon>
  </Button>;
