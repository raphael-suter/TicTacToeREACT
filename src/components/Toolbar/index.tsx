import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ title, score, children }: Props) => (
  <header className='toolbar'>
    <h1>{title}</h1>
    <p>{score}</p>
    <div>
      {children}
    </div>
  </header>
);
