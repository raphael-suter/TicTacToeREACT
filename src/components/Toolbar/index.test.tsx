import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Toolbar from '.';

describe('Toolbar', () => {
  test('Renders correctly.', () => {
    const wrapper = shallow(<Toolbar title='Title' score='Score' children={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
