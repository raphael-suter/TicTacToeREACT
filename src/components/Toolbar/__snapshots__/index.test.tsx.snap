// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Renders correctly. 1`] = `
<styled.header>
  <styled.h1>
    Title
  </styled.h1>
  <styled.p>
    Score
  </styled.p>
  <styled.div />
</styled.header>
`;
